### PR TITLE
TASK-7

### PR DESCRIPTION
**Pull request by skyRoma**
- [x]  - authorization-service is added to the repo, has correct basicAuthorizer lambda and correct serverless.yaml file
- [x]  - import-service serverless.yaml file has authorizer configuration for the importProductsFile lambda. Request to the importProductsFile lambda should work only with correct authorization_token being decoded and checked by basicAuthorizer lambda. Response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if Authorization header is not provided.
- [x]  - update client application to send Authorization: Basic authorization_token header on import. Client should get authorization_token value from browser localStorage https://developer.mozilla.org/ru/docs/Web/API/Window/localStorage authorization_token = localStorage.getItem('authorization_token')
Additional (optional) tasks: 
- [x] - Client application should display alerts for the responses in 401 and 403 HTTP statuses. This behavior should be added to the nodejs-aws-fe-main/src/index.tsx file

**FE PR**: https://github.com/skyRoma/nodejs-aws-fe/pull/6
**FE APP**: https://d6zd7boq0dknk.cloudfront.net
**Token**: c2t5Um9tYTpURVNUX1BBU1NXT1JE
**Header**: Basic c2t5Um9tYTpURVNUX1BBU1NXT1JE
**Import url**: https://00itcm6s0m.execute-api.eu-west-1.amazonaws.com/dev/import?name=test.csv

image

### CODE REVIEW

**Comment by skyRoma** [comment]
**!Important** Deployment incident
I had to remove all the stacks and even the old functionality got lost. I'm working on a fix.

**Comment by skyRoma** [comment]
> **!Important** Deployment incident
> I had to remove all the stacks and even the old functionality got lost. I'm working on a fix.

Fixed

**Comment by maksumov** [comment]
image

**Comment by skyRoma** [comment]
@maksumov  this is expected, since the error is 500 (and the requirements don't say to handle it for ApiGateway), because the header consists of two words (Basic+ token from local storage). The requirements say about the entire request header, not only about the token part.
image

If I remove the header from the code I have (as described in requirements):

image
image

**Comment by maksumov** [comment]
I believe that three types of responses should be received from the authorizer: with the status codes 200, 401, or 403. If the request has an incorrect header, you can also return the 400 status, but it seems to me that the 401 status is suitable for this case. In fact, the authorization header is not provided. What is sent is garbage

File: import-service/serverless.ts [hunk 1]
```diff
@@ -85,6 +101,36 @@ const serverlessConfiguration: Serverless = {
...
     },
   },
+  resources: {
+    Resources: {
+      GatewayResponseAccessDenied: {
+        Type: 'AWS::ApiGateway::GatewayResponse',
+        Properties: {
+          ResponseParameters: {
+            'gatewayresponse.header.Access-Control-Allow-Origin': "'*'",
+            'gatewayresponse.header.Access-Control-Allow-Headers': "'*'",
```
Comment (maksumov) [inline import-service/serverless.ts:111]: Try to change **gatewayresponse.header.Access-Control-Allow-Headers** by **gatewayresponse.header.Access-Control-Allow-Credentials**

Comment (skyRoma) [reply import-service/serverless.ts:111]: it handled this way:
https://github.com/skyRoma/nodejs-aws-be/pull/5/commits/2b376de08eed6f945c347e76d1f1c0ba33e691d8

**Comment by skyRoma** [comment]
> I believe that three types of responses should be received from the authorizer: with the status codes 200, 401, or 403. If the request has an incorrect header, you can also return the 400 status, but it seems to me that the 401 status is suitable for this case. In fact, the authorization header is not provided. What is sent is garbage

just in case, handled:
https://github.com/skyRoma/nodejs-aws-be/pull/5/commits/2b376de08eed6f945c347e76d1f1c0ba33e691d8

**Review by spitaI — APPROVED** [review]

**Review by maksumov — APPROVED** [review]

**Pull request merged**